activation/nipost.go

```diff
@@ -407,57 +407,72 @@ func membersContainChallenge(members []types.Member, challenge types.Hash32) (ui
 	return 0, fmt.Errorf("challenge is not a member of the proof")
 }
 
-// TODO(mafa): remove after epoch 4 ends; https://github.com/spacemeshos/go-spacemesh/issues/4968
-func (nb *NIPostBuilder) addPoet111ForPubEpoch4(ctx context.Context) error {
-	// because PoET 111 had a hardware issue when challenges for round 3 were submitted, no node could submit to it
-	// 111 was recovered with the PoET 110 DB, so all successful submissions to 110 should be able to be fetched from there as well
-
-	client111, ok := nb.poetProvers["https://poet-111.spacemesh.network"]
+func (nb *NIPostBuilder) addPoETMitigation(ctx context.Context, from, to string, pubEpoch types.EpochID) error {
+	clientTo, ok := nb.poetProvers[to]
 	if !ok {
 		// poet 111 is not in the list, no action necessary
 		return nil
 	}
 
-	nb.log.Info("pub epoch 4 mitigation: PoET 111 is in the list of PoETs, adding it to the state as well")
-	client110 := nb.poetProvers["https://poet-110.spacemesh.network"]
+	nb.log.With().Info("poet mitigation: target PoET is in the list of PoETs, adding it to the state as well",
+		log.String("state_from", from),
+		log.String("target_poet", to),
+		log.Uint32("pub_epoch", pubEpoch.Uint32()),
+	)
+	clientFrom := nb.poetProvers[from]
```

Comment:
I think it should be checked if `from` is in `nb.poetProvers` to avoid panicking 1 line below (which could happen if somebody removed poet-110 from their poets' configuration).